### DISCUSSION TITLE
Multiple instances

### DISCUSSION THREAD

**Discussion by listentolist** [leo-arch/clifm#212] [Ideas]
As written in the wiki about multiple instances there is the problem that the last closed instance overwrites the history files produced by the previous ones. This makes using multiple instances very inconvenient. What would be the best solution for this problem?

**Comment by leo-arch** [comment]
Hi @listentolist. Yes, this is currently the behavior of the directory history (not though of the commands history).

The solution would be to append new entries to the history file every time the current directory is changed, instead of doing it all at once at exit. I'll take a look at it.

Thanks for reporting.

**Reply by leo-arch** [reply]
@listentolist, done! Give it a try and let me know if it works as intended.

NOTE: synchronization between multiple instances happen at startup, for each instance keeps a private cache of the directory history, so that it won't be aware of changes made by other instances until next start.

**Reply by listentolist** [reply]
@leo-arch Thank you! When using the directory jumper I still have kind of the same problem. Maybe I should have been a bit more specific about it. I thought of the jump list as a part of the history files. Here is a minimal example:

Let's say I've got the directories a and b.

1. start a new clifm instance and change to directory a -> a is shown in the jump list
2. start a new clifm instance in another terminal and change to directory b -> a is not shown in the jump list but b is
3. quit the first instance
4. quit the second instance
5. start a new clifm instance -> b is shown in the jump list, but a is not

The best would be if at any point the jump lists of the different instances are identical and that we do not lose any entry on exit or startup.

**Reply by leo-arch** [reply]
I committed a fix like an hour ago that is intended to address this exact issue.

Please give it a try now.

**Reply by listentolist** [reply]
I just noticed that you added another commit that improves the synchronization between the jump database and the dirs history. That solves the problem. Thank you!
